src/MediaUploader/utils.js

```diff
@@ -0,0 +1,64 @@
+import {includes} from 'lodash'
+
+// upload to a presigned s3 url
+function makeS3Request(data, file) {
+  return new Promise((resolve, reject) => {
+    const xhr = new XMLHttpRequest()
+
+    xhr.onreadystatechange = () => {
+      if (xhr.readyState === 4) {
+        if (xhr.status === 200) {
+          resolve(data.url)
+        } else {
+          reject({
+            status: xhr.status,
+            statusText: xhr.statusText,
+          })
+        }
+      }
+    }
+
+    xhr.open('PUT', data.signed_request)
```

Comment:
`data` and it's properties may be too specific and need to be made more general for larger use, but for our purposes it works fine